### ISSUE TITLE
Using static Type<T> with constraints produces incorrect program instead of build error.

### ISSUE THREAD

**Issue by zlatanov** [dotnet/roslyn#30726]
**Version Used**: 15.8.7

**Steps to Reproduce**:

```c#
using static Test<System.String>;

public static class Test<T> where T : struct
{
    public static void Run( T item ) { }
}

class Program
{
    static void Main( string[] args )
    {
        Run( "This should not compile." );
    }
}
```

**Expected Behavior**:

The compiler should issue an error:
**CS0453**: The type 'string' must be a non-nullable value type in order to use it as parameter 'T' in the generic type or method 'Test<T>'

**Actual Behavior**:
The program compiles. When you run the program we get:
**System.TypeLoadException:** 'GenericArguments[0], 'System.String', on 'Test`1[T]' violates the constraint of type parameter 'T'.'

The compiled IL code is:

```IL
IL_0000: ldstr "This should not compile."
IL_0005: call void class Test`1<string>::Run(!0)
```

**Comment by RikkiGibson** [comment]
Seems like the diagnostic should reference the `using static` directive itself?
